table/block_based_table_reader.cc

```diff
@@ -813,7 +813,10 @@ Status BlockBasedTable::Open(const ImmutableCFOptions& ioptions,
       CachableEntry<IndexReader> index_entry;
       unique_ptr<InternalIterator> iter(
           new_table->NewIndexIterator(ReadOptions(), nullptr, &index_entry));
-      index_entry.value->CacheDependencies(pin);
+      assert(index_entry.value != nullptr || !iter->status().ok());
```

Comment:
Can you add some in-line comments of why assert on this disjunction make sense? It is at least not clear to me.

Reply:
Sure.